src/common/src/array/data_chunk.rs

```diff
@@ -344,18 +344,25 @@ impl DataChunk {
         Ok(outputs)
     }
 
-    /// Compute hash values for each row.
+    /// Compute hash values for each row. The number of the returning `HashCodes` is `self.capacity()`.
+    /// When `skip_invisible_row` is true, the `HashCode` for the invisible rows is arbitrary.
```

Comment:
What about having two methods, one for skipping, another for not skipping, the skipping one returns `Vec<Option<HashCode<H>>>`? Returning arbitrary hash code looks unsafe to me.

Reply:
I want to remove the `skip_invisible_row = true` later and I think it is not needed and the current state is just to prevent the potential bugs.